services/mixpanel.ts

```diff
@@ -0,0 +1,30 @@
+import mixpanel from "mixpanel-browser";
+
+export class MixpanelTracking {
+
+    private static _instance: MixpanelTracking;
+
+    public static getInstance(): MixpanelTracking {
+        if (MixpanelTracking._instance == null) {
+            return (MixpanelTracking._instance = new MixpanelTracking());
+        }
+        return this._instance;
+    }
+
+    public constructor() {
+        if (MixpanelTracking._instance) {
+            throw new Error('Error: Instance creation of MixpanelTracking is not allowed. Use Mixpanel.getInstance instead.')
+        }
+        mixpanel.init(process.env.NEXT_PUBLIC_MIXPANEL_TOKEN || '', {
```

Comment:
this token would be visible in browser na? its public?

Reply:
![image](https://github.com/avantifellows/gurukul/assets/107760942/67224e58-f265-4ee5-b3ec-d0d9c6b5246b)